### PR TITLE
#159965304 enable social login 

### PR DESCRIPTION
**Pull request by pabishai**
**What does this PR do?**
Enable user login using Facebook, Google and Twitter

**Description of Task to be completed?**
Have OAuth and OAuth2 authentication working in Django

**How should this be manually tested?**
Run `git clone https://github.com/andela/ah-magnificent6.git` from your local machine to clone the repo. 
Run `git checkout ft-enable-social-login-159965304 ` to go to the relevant branch. 
Run`python3 -m venv venv` and `source venv/bin/activate` to activate a virtual environment. 
Run `pip install -r requirements.txt` to install the required packages.
Create a `.env` file from the `.env-sample` and add the relevant environmental variables
Run `python manage.py migrate` to run migrations. This is necessary
Run `python manage.py runserver` to start the app.
Test the endpoint `` using an API testing environment like postman
The expected payloads are in the format:
`POST /api/users/oauth/`
Twitter: 
```
{
    "provider": "twitter",
    "access_token": "valid_access_token",
    "access_token_secret": "valid_access_token_secret"
}
```

Facebook: 
```
{
    "provider":"facebook",
    "access_token":"valid_facebook_access_token"
}
```

Google: 
```
{
    "provider":"google-oauth2",
    "access_token":"valid_google_access_token"
}
```
The access tokens can be generated from the provider's developer pages:
[Twitter](https://developer.twitter.com/en/dashboard)
[Google](https://developers.google.com/oauthplayground/)
[Facebook](https://developers.facebook.com/tools/explorer/)

**What are the relevant pivotal tracker stories?**
[#159965296](https://www.pivotaltracker.com/story/show/#159965296)

**Screenshots**
Successfull OAuth feedback on Postman
screen shot 2018-09-11 at 20 47 14

### CODE REVIEW

**Comment by micahoriaso** [comment]
This looks great to me. It works as expected

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ install:
   - pip install coverage
 services:
   - postgresql
+  
+addons:
+  postgresql: "9.4"
```
Comment (parseen254) [inline .travis.yml:12]: @PAbishai was this necessary?

File: requirements.txt [hunk 2]
```diff
@@ -28,3 +28,9 @@ uritemplate==3.0.0
 urllib3==1.23
 whitenoise==4.0
 django-rest-swagger==2.2.0
+python-social-auth==0.3.6
+six==1.10.0
+social-auth-app-django==2.1.0
+social-auth-core==1.7.0
+whitenoise==4.0
```
Comment (parseen254) [inline requirements.txt]: whitenose and swagger are already present on lines 29 and 30 respectively.

**Review by parseen254 — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (pabishai) [reply .travis.yml:12]: I was getting this error from Travis "ProgrammingError: syntax error at or near "WITH ORDINALITY" and the solution was to enable use of Postgres 9.4 or higher

File: requirements.txt (hunk 2 above)
Comment (pabishai) [reply requirements.txt]: Let me ammend that

Comment (pabishai) [reply requirements.txt]: @parseendavid All done

File: .travis.yml (hunk 1 above)
Comment (parseen254) [reply .travis.yml:12]: Okay

**Review by parseen254 — APPROVED** [review]

**Pull request merged**